packages/plugins/typescript/document-nodes/src/index.ts

```diff
@@ -0,0 +1,122 @@
+import { print } from 'graphql/language/printer';
+
+export type NamingConvention = 'lowerCamelCase' | 'UpperCamelCase' | 'UPPER_CASE';
+
+function upperFirst(str: string): string {
+  return str[0].toUpperCase() + str.slice(1);
+}
+
+function camelize(str: string): string {
+  return str
+    .replace(/([a-z])([A-Z])/g, '$1 $2')
+    .split(/[\x00-\x2f\x3a-\x40\x5b-\x60\x7b-\x7f]+/)
+    .reduce((result, word, index) => {
+      word = word.toLowerCase();
+      return result + (index ? upperFirst(word) : word);
+    }, '');
+}
+
+function upperize(str: string): string {
+  return str
+    .replace(/([a-z])([A-Z])/g, '$1 $2')
+    .replace(/[\x00-\x2f\x3a-\x40\x5b-\x60\x7b-\x7f]+/g, '_')
+    .toUpperCase();
+}
+
+function useNamingConvention(str: string, namingConvention?: NamingConvention): string {
+  if (namingConvention === 'lowerCamelCase') {
+    return camelize(str);
+  } else if (namingConvention === 'UpperCamelCase') {
+    return upperFirst(camelize(str));
+  } else if (namingConvention === 'UPPER_CASE') {
+    return upperize(str);
+  } else {
+    return str;
+  }
+}
+
+export interface TypeScriptDocumentNodesPluginConfig {
+  /**
+   * @name namingConvention
+   * @type string
+   * @default false
+   * @description Generates variable names in choosen naming convention
+   * (lowerCamelCase, UpperCamelCase or UPPER_CASE)
+   *
+   * @example
+   * ```yml
+   *  generates: src/api/user-service/queries.ts
+   *  documents: src/api/user-service/queries.graphql
+   *  plugins:
+   *    - graphql-codegen-typescript-document-nodes
+   *  config:
+   *    namingConvention: lowerCamelCase
+   * ```
+   */
+  namingConvention?: NamingConvention;
+  /**
+   * @name namePrefix
+   * @type string
+   * @default ''
+   * @description Adds prefix to the name
+   *
+   * @example
+   * ```yml
+   *  generates: src/api/user-service/queries.ts
+   *  documents: src/api/user-service/queries.graphql
+   *  plugins:
+   *    - graphql-codegen-typescript-document-nodes
+   *  config:
+   *    namePrefix: 'gql'
+   * ```
+   */
+  namePrefix?: string;
+  /**
+   * @name nameSuffix
+   * @type string
+   * @default ''
+   * @description Adds suffix to the name
+   *
+   * @example
+   * ```yml
+   *  generates: src/api/user-service/queries.ts
+   *  documents: src/api/user-service/queries.graphql
+   *  plugins:
+   *    - graphql-codegen-typescript-document-nodes
+   *  config:
+   *    nameSuffix: 'Query'
+   * ```
+   */
+  nameSuffix?: string;
+}
+
+export const plugin: PluginFunction<TypeScriptDocumentNodesPluginConfig> = (schema: GraphQLSchema, documents: Types.DocumentFile[], config: TypeScriptDocumentNodesPluginConfig): string => {
+  const { namingConvention, namePrefix = '', nameSuffix = '' } = config;
+  const content = documents
+    .filter(documentFile => documentFile.filePath.length > 0)
+    .map(documentFile =>
+      documentFile.content.definitions
+        .filter((d: OperationDefinitionNode) => d.name && d.name.value)
```

Comment:
Maybe add `console.warn` here to notify about the ignored operations?

Reply:
I'm not convinced. This pattern is copy-pasted from `graphql-files-modules`. Should I change it in all existing plugins?

Reply:
@dotansimha I double checked this code. It works like additional sanitisation and filtering it is very easy but making some warning on missing value might be more difficult and probably unnecessary.